src/Microsoft.PowerShell.Commands.Utility/commands/utility/WebCmdlet/Common/WebRequestPSCmdlet.Common.cs

```diff
@@ -1259,13 +1259,13 @@ internal virtual HttpResponseMessage GetResponse(HttpClient client, HttpRequestM
                     {
                         WebSession.MaximumRedirection--;
                     }
-                    // For selected redirects that used POST, GET must be used with the
-                    // redirected Location.
-                    // Since GET is the default; POST only occurs when -Method POST is used.
-                    if (Method == WebRequestMethod.Post && IsRedirectToGet(response.StatusCode))
+
+                    // For selected redirects, GET must be used with the redirected Location.
+                    if (req.Method == HttpMethod.Post && IsRedirectToGet(response.StatusCode))
```

Comment:
Could you please rename `req` to `currentRequest`. This will make the code more readable.